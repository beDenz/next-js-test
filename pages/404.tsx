import Error from 'next/error'

export default function NotFound(): React.ReactNode {
  return <Error statusCode={404} title="Page not found"></Error>
}
