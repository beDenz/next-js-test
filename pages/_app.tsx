import '../styles/global.scss'
import type { AppProps } from 'next/app'

function App({ Component, pageProps }: AppProps): React.ReactNode {
  return <Component {...pageProps} />
}

export default App
