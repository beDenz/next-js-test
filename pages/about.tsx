import { Layout } from 'components/layout/Layout'

function About({ menuItem, aboutPageMeta, title, text, ver }): React.ReactNode {
  return (
    <Layout menuItem={menuItem} meta={aboutPageMeta}>
      <h1>{title}</h1>
      <p>{text}</p>
      <p>ver {ver}</p>
    </Layout>
  )
}

export async function getStaticProps() {
  const res = await fetch(process.env.ABOUT_PAGE)

  const result = await res.json()

  const { menuItem, aboutPageMeta, title, text, ver } = result

  return {
    props: {
      menuItem,
      aboutPageMeta,
      title,
      text,
      ver,
    },
    revalidate: 60,
  }
}

export default About
