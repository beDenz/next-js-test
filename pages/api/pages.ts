// Next.js API route support: https://nextjs.org/docs/api-routes/introduction

import { NextApiRequest, NextApiResponse } from 'next'

const menuItem = [
  {
    id: 1,
    link: '/',
    title: 'home',
  },
  {
    id: 2,
    link: '/pages',
    title: 'pages',
  },
  {
    id: 3,
    link: '/about',
    title: 'about',
  },
  {
    id: 4,
    link: '/ssr',
    title: 'ssr',
  },
  {
    id: 5,
    link: '/',
    title: 'login',
  },
]


const pagesIds = [
  {
    id: 1,
    link: '/pages/1',
    title: '/pages/1'
  },
  {
    id: 2,
    link: '/pages/2',
    title: '/pages/2'
  },
  {
    id: 3,
    link: '/pages/3',
    title: '/pages/3'
  },
  {
    id: 4,
    link: '/pages/4',
    title: '/pages/4'
  },
  {
    id: 5,
    link: '/pages/5',
    title: '/pages/5'
  },
]

const pagesPageMeta = {
  title: 'Pages Page',
  description: 'pages next js page info',
  keywords: 'bla bla bla',
}

const title = 'Pages Page'



const pagesPath = [
  {
    params: {
      id: '1'
    },
  },
  {
    params: {
      id: '2'
    }
  }, {
    params: {
      id: '3'
    }
  },
  {
    params: {
      id: '4'
    }
  },
  {
    params: {
      id: '5'
    }
  }
]



const handler = (req: NextApiRequest, res: NextApiResponse) => {
  res.status(200).json({ menuItem, pagesPageMeta, title, pagesIds, pagesPath })
}

export default handler
