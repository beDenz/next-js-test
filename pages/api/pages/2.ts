/* eslint-disable */
import { NextApiRequest, NextApiResponse } from 'next'

const menu = [
  {
    id: 1,
    link: '/',
    title: 'home',
  },
  {
    id: 2,
    link: '/pages',
    title: 'pages',
  },
  {
    id: 3,
    link: '/about',
    title: 'about',
  },
  {
    id: 4,
    link: '/ssr',
    title: 'ssr',
  },
  {
    id: 5,
    link: '/',
    title: 'login',
  },
]

const meta = {
  title: 'Page 2 title',
  description: 'Page 2 description',
  keywords: 'page 2 page 2 page 2',
}

const title = 'Page #2'

const ver = 1

const handler = (req: NextApiRequest, res: NextApiResponse) => {
  res.status(200).json({ menu, meta, title, ver })
}

export default handler
