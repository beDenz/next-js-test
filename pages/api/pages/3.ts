/* eslint-disable */
import { NextApiRequest, NextApiResponse } from 'next'

const menu = [
  {
    id: 1,
    link: '/',
    title: 'home',
  },
  {
    id: 2,
    link: '/pages',
    title: 'pages',
  },
  {
    id: 3,
    link: '/about',
    title: 'about',
  },
  {
    id: 4,
    link: '/ssr',
    title: 'ssr',
  },
  {
    id: 5,
    link: '/',
    title: 'login',
  },
]

const meta = {
  title: 'Page 3 title',
  description: 'Page 3 description',
  keywords: 'page 3 page 3 page 3',
}

const title = 'Page #3'

const ver = 1

const handler = (req: NextApiRequest, res: NextApiResponse) => {
  res.status(200).json({ menu, meta, title, ver })
}

export default handler
