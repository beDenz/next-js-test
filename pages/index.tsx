import { Layout } from '../components/layout/Layout'

export const Home = ({ menuItem, mainPageMeta }): JSX.Element => {
  return (
    <Layout menuItem={menuItem} meta={mainPageMeta}>
      <div>Main Page</div>
    </Layout>
  )
}

export async function getStaticProps() {
  const res = await fetch(process.env.MAIN_PAGE)

  const result = await res.json()

  const { menuItem, mainPageMeta } = result

  return {
    props: {
      menuItem,
      mainPageMeta,
    },
  }
}

export default Home
