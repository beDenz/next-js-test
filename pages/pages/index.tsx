import { Layout } from 'components/layout/Layout'
import { MenuItem, Meta } from 'types/types'
import Link from 'next/link'

type PagesIdItem = {
  id: string
  link: string
  title: string
}

type PagesProps = {
  menuItem: MenuItem[]
  pagesPageMeta: Meta
  title: string
  pagesIds: PagesIdItem[]
}

const Pages = ({ menuItem, pagesPageMeta, title, pagesIds }: PagesProps) => {
  return (
    <Layout menuItem={menuItem} meta={pagesPageMeta}>
      <h1>{title}</h1>
      <ul>
        {pagesIds.map((item) => (
          <li key={item.id}>
            <Link href={item.link}>{item.title}</Link>
          </li>
        ))}
      </ul>
    </Layout>
  )
}

export async function getStaticProps() {
  console.log('!!!Pages getStaticProps!!!')

  const res = await fetch(process.env.PAGES_PAGE)

  const result = await res.json()

  const { menuItem, pagesPageMeta, title, pagesIds } = result

  return {
    props: { menuItem, pagesPageMeta, title, pagesIds },
  }
}

export default Pages
