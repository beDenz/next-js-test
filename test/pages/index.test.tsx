import React from 'react'
import { render } from '../testUtils'
// import { render, fireEvent } from '../testUtils'
import { Home } from 'pages/index'

const menu = [
  {
    id: 1,
    link: '/',
    title: 'home',
  },
  {
    id: 2,
    link: '/pages',
    title: 'pages',
  },
  {
    id: 3,
    link: '/about',
    title: 'about',
  },
  {
    id: 4,
    link: '/ssr',
    title: 'ssr',
  },
  {
    id: 5,
    link: '/',
    title: 'login',
  },
]

const meta = {
  title: 'Main Page',
  description: 'main next js page info',
  keywords: 'bla-bla-bla',
}

const title = 'Main Page'

const text = `Lorem, ipsum dolor sit amet consectetur adipisicing elit. Maxime dolorem
unde, corporis eligendi inventore blanditiis autem natus minima quasi
sunt, esse eaque! Soluta, cum tempora. Maiores numquam magnam eos culpa.`

const ver = '1'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <Home menu={menu} meta={meta} title={title} text={text} ver={ver} />,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })

  // it('clicking button triggers alert', () => {
  //   const { getByText } = render(<Home />, {})
  //   window.alert = jest.fn()
  //   fireEvent.click(getByText('Test Button'))
  //   expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
  // })
})
