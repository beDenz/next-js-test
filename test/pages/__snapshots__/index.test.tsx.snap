// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <header
      class="header"
    >
      <div
        class="logo"
      >
        <span>
          Logo
        </span>
      </div>
      <ul
        class="navMenu"
      >
        <li
          class="nav-menu-item"
        >
          <a
            href="/"
          >
            home
          </a>
        </li>
        <li
          class="nav-menu-item"
        >
          <a
            href="/pages"
          >
            pages
          </a>
        </li>
        <li
          class="nav-menu-item"
        >
          <a
            href="/about"
          >
            about
          </a>
        </li>
        <li
          class="nav-menu-item"
        >
          <a
            href="/ssr"
          >
            ssr
          </a>
        </li>
        <li
          class="nav-menu-item"
        >
          <a
            href="/"
          >
            login
          </a>
        </li>
      </ul>
    </header>
    <main>
      <h1>
        Main Page
      </h1>
      <p>
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Maxime dolorem
unde, corporis eligendi inventore blanditiis autem natus minima quasi
sunt, esse eaque! Soluta, cum tempora. Maiores numquam magnam eos culpa.
      </p>
      <p>
        ver 1
      </p>
    </main>
    <footer />
  </div>
</DocumentFragment>
`;
